src/httpserver.cpp

```diff
@@ -419,6 +419,15 @@ bool InitHTTPServer()
     int workQueueDepth = std::max((long)gArgs.GetIntArg("-rpcworkqueue", DEFAULT_HTTP_WORKQUEUE), 1L);
     LogPrintfCategory(BCLog::HTTP, "creating work queue of depth %d\n", workQueueDepth);
 
+#if LIBEVENT_VERSION_NUMBER >= 0x02020001
+    if (event_get_version_number() >= 0x02020001) {
+        // Limit the maximum number of open connections to prevent exhausting
+        // the file descriptor limit. When the http server gets overwhelmed it
+        // will respond with 503 Service Unavailable.
+        evhttp_set_max_connections(http, workQueueDepth * 2);
```

Comment:
I am not sure this actually limits us within the file descriptor limit. For example if you run:

```bash
$ ulimit -n 175
$ src/bitcoind
Warning: Reducing -maxconnections from 125 to 15, because of system limitations.
Bitcoin Core starting
```

I think (but have not tested) that if I then recieve 15 inbound connections I would could still run out of file descriptors for the http server?

Perhaps if we also modify this line:

https://github.com/bitcoin/bitcoin/blob/51c050787fd6bcd016969dd7e245818ebd110b67/src/init.cpp#L943

...to consider the workqueue depth too, then it could work?